modules/local/append_profiles/main.nf

```diff
@@ -0,0 +1,40 @@
+process APPEND_PROFILES {
+    tag "Append additional reference profiles"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/csvtk:0.22.0--h9ee0642_1' :
+        'biocontainers/csvtk:0.22.0--h9ee0642_1' }"
+
+    input:
+    path(reference_profiles)
+    path(additional_profiles)
+
+    output:
+    path("profiles_ref.tsv")
+
+    script:
+    """
+    # Function to get the header of the files, handling gzipped files
+    get_header() {
+        if [ "\${1##*.}" = "gz" ]; then
+            zcat "\$1" | head -n 1
+        else
+            head -n 1 "\$1"
+        fi
+    }
+
+    # Compare headers and exit if they do not match
+    ref_headers=\$(get_header "${reference_profiles}")
+    add_headers=\$(get_header "${additional_profiles}")
+
+    if [ "\$ref_headers" != "\$add_headers" ]; then
+        echo "Error: Column headers do not match between reference_profiles and --db_profiles."
+        exit 1
+    fi
+
+    # Merge profiles ensuring only unique samples are added
+    csvtk concat -t ${reference_profiles} ${additional_profiles} > profiles.tsv
+    csvtk uniq -t -f sample_id profiles.tsv > profiles_ref.tsv
```

Comment:
Thanks so much for adding this in to check for and remove duplicate samples.

One question I have, the docs for csvtk (https://bioinf.shenwei.me/csvtk/usage/#uniq) say that it applies unique without sorting. Does this mean that it will only remove duplicate samples if they happen right next to each other? Could you check?

Reply:
Great catch, Aaron! You're absolutely right. I've added a `csvtk sort` to ensure duplicate sample_ids are grouped correctly for proper removal by `csvtk uniq`. I also updated the tests to account for the sorting and adjusted the order of the duplicate sample_ids so they are no longer adjacent.
Changes can be found here: [9780166](https://github.com/phac-nml/gasnomenclature/pull/29/commits/9780166b3250c45773f2bde5f720fe9f391ea488)
And test updates here: [7d2f30a](https://github.com/phac-nml/gasnomenclature/pull/29/commits/7d2f30ab31dafefcbe643ace3c136c73813cb3c1)

Reply:
Thank you for all the great work you've put into this :)